examples/gno.land/r/gov/dao/dao.gno

```diff
@@ -8,38 +8,58 @@ import (
 	pproposal "gno.land/p/gov/proposal"
 )
 
-var proposals = make([]*proposal, 0)
+var (
+	proposals = make([]*proposal, 0)
+	members   = make([]std.Address, 0) // XXX: these should be pointers to avoid data duplication. Not possible due to VM bugs
+)
+
+const (
+	msgMissingExecutor = "missing proposal executor"
+	msgPropExecuted    = "prop already executed"
+	msgPropExpired     = "prop is expired"
+	msgPropInactive    = "prop is not active anymore"
+	msgPropActive      = "prop is still active"
+	msgPropNotAccepted = "prop is not accepted"
 
-// XXX var members ...
+	msgCallerNotAMember = "caller is not member of govdao"
+	msgProposalNotFound = "proposal not found"
+)
 
 type proposal struct {
-	idx      int
 	author   std.Address
 	comment  string
 	executor pproposal.Executor
-	// XXX: make "accepted" and "finished" managed by an interface that can have various voting implementations
-	accepted bool
-	finished bool
+	voter    Voter
+	executed bool
+	voters   []std.Address // XXX: these should be pointers to avoid data duplication. Not possible due to VM bugs.
```

Comment:
Pointer slices should work; AVL trees not.

Reply:
we'll need to switch to an avl.tree or another storage-efficient alternative quickly.

Reply:
The problem is not in the slice itself, but in the pointers to structs operations, like trying to check if an address is already on the slice, you have to convert the new address to a pointer and compare it. That is the part that doesn't work.